browser/resources/settings/getting_started_page/brave_account_create_dialog.ts

```diff
@@ -0,0 +1,198 @@
+        height: 4px;
+        transition: 750ms;
+        width: 376px;
+      }
+
+      .strength {
+        background-color: var(--primary-color);
+        border-radius: var(--leo-radius-m);
+        height: 100%;
+        transition: 750ms;
+        width: calc(1% * var(--strength));
+      }
+
+      .text {
+        color: var(--primary-color);
+        font: var(--leo-font-small-regular);
+        transition: 750ms;
+      }
+    `
+  }
+
+  override render() {
+    return html`
+      <div class="bar">
+        <div class="strength" style="--strength: ${this.strength}"></div>
+      </div>
+      <div class="text">
+        ${this.i18n(`braveAccountPasswordStrengthMeter${this.category}`)}
+      </div>
+    `
+  }
+
+  static override get properties() {
+    return {
+      category: { type: String, reflect: true },
+      strength: { type: Number },
+    }
+  }
+
+  override updated(changedProperties: Map<PropertyKey, unknown>) {
+    if (changedProperties.has('strength')) {
+      this.category = this.strength < 60
+        ? 'Weak'
+        : this.strength < 100
+          ? 'Medium'
+          : 'Strong'
+    }
+  }
+
+  protected category: string = 'Weak'
+  protected strength: number = 0
+}
+
+declare global {
+  interface HTMLElementTagNameMap {
+    'password-strength-meter': PasswordStrengthMeter
+  }
+}
+
+customElements.define(
+  PasswordStrengthMeter.is, PasswordStrengthMeter)
+
+export class SettingsBraveAccountCreateDialogElement extends CrLitElement {
+  static get is() {
+    return 'settings-brave-account-create-dialog'
+  }
+
+  static override get styles() {
+    return getCss()
+  }
+
+  override render() {
+    return getHtml.bind(this)()
+  }
+
+  static override get properties() {
+    return {
+      email: { type: String },
+      isAccountNameValid: { type: Boolean },
+      isCheckboxChecked: { type: Boolean },
+      isEmailBraveAlias: { type: Boolean },
+      isEmailValid: { type: Boolean },
+      password: { type: String },
+      passwordConfirmation: { type: String },
+      passwordStrength: { type: Number },
+    }
+  }
+
+  protected onEmailInput(detail: { value: string }) {
+    this.email = detail.value
+    this.isEmailValid = emailRegexp.test(this.email)
+    this.isEmailBraveAlias = this.email.endsWith('@bravealias.com')
+  }
+
+  protected onAccountNameInput(detail: { value: string }) {
+    this.isAccountNameValid = detail.value.length !== 0
+  }
+
+  protected onPasswordInput(detail: { value: string }) {
+    this.password = detail.value
+    this.browserProxy.getPasswordStrength(this.password).then(
+      (passwordStrength: number) => this.passwordStrength = passwordStrength
+    )
+  }
+
+  protected onConfirmPasswordInput(detail: { value: string }) {
+    this.passwordConfirmation = detail.value
+  }
+
+  protected onCheckboxChanged(detail: { checked: boolean }) {
+    this.isCheckboxChecked = detail.checked
+  }
+
+  // TODO(sszaloki): we should consider exporting `noChange`
```

Comment:
out of interest, why would we need `noChange` here?

Reply:
Basically, when there's a matching password in the password confirmation field, we show the checkmark and the text `Passwords match`. Now, if the user presses `Cmd` + `Backspace` on the text in the password confirmation field, we should just hide the dropdown and should not recalculate which icon or text to show — since the password confirmation field became empty, and there's nothing in there we should try to match against the field above. That's an edge-case here, but if we don't do anything, there's a visual artifact: the icon changes to the red triangle, while the text changes to `Passwords don't match`, and then the dropdown disappears — which looks rookie. 🙂 

The proper way to handle that would be via signaling with `noChange` that we don't need a change in the DOM if `this.passwordConfirmation.length === 0`. Alternatively, there is this hack we have in `brave_account_create_dialog.ts`, but it's far from ideal, as the TS file is required to know about style-related stuff (i.e. name of icons).